code/0010-define-calculations-of-existential-deposit.md

```diff
@@ -0,0 +1,149 @@
+initial definition of all ED values on our chain while also allowing for 
+extensibility in the future.
+
+## Requirements
+
+* ED values for all tokens supported by the chain are available AND consistent
+  across our runtime, node, and front-end applications.
+
+* When the ED of token A is swapped into token B, the new balance of token B is
+  substantial for ED. In other words, a user cannot "delete" their balance when 
+  they convert it into another token.
+
+* The implementation of our ED value retrieval should allow for the overwriting
+  of the standard ED value defined in this document.
+
+## Method
+
+### Initial Standard ED Value
+
+With the following definitions: 
+
+* `local_ed` - The functional ED of a asset on our chain.
+
+* `native_token_ed` - The ED of our chains native asset.
+
+* `foreign_asset_ratio` - The ratio of 'one in the foreign asset' over 'one in 
+our native asset'
+
+The ED of any asset should be defined as follows:
+```
+local_ed = native_token_ed * foreign_asset_ratio
+```
+
+This method of setting ED ensures requirement 2.
+
+Example
+```
+token_a_local_ed = 100_000_000
+token_a_asset_ratio = Ratio(1_000_000_000, 1_000_000_000_000)
+token_b_local_ed = 100_000
+token_a_asset_ratio = Ratio(1_000_000, 1_000_000_000_000)
+native_token_ed = 100_000_000_000
+
+fn native_to_token(native_balance, token_asset_ratio) {
+  native_balance * token_asset_ratio
+}
+
+fn token_to_native(token_balance, token_asset_ratio) {
+  token_balance / token_asset_ratio
+}
+
+token_a_ed_as_native = token_to_native(token_a_local_ed, token_a_asset_ratio)
+token_b_ed_as_native = token_to_native(token_b_local_ed, token_b_asset_ratio)
+native_as_token_b = native_to_token(native_token_ed, token_b_asset_ratio)
+token_a_as_token_b = native_to_token(token_a_ed_as_native, token_b_asset_ratio)
+
+assert(token_a_ed_as_native == native_token_ed)
+assert(token_b_ed_as_native == native_token_ed)
+assert(token_a_ed_as_native == token_b_ed_as_native)
+assert(native_as_token_b == token_a_as_token_b)
+```
+
+### Overwriting & Asset Registry Storage
+
+For a token to have its ED be correctly calculated, and therefore to exists on 
+chain, it should have its `foreign_asset_ratio` defined and stored.
+
+If the need arises, an optional ED value can be provided to an asset within
+Asset Registry. This will have priority over the default ED calculation and will
+be evaluated as is.
+
+## Implementation
+
+### Updates to the Functionality of `multi_existential_deposits`
+
+The current implementation of `multi_existential_deposits` does the following:
+  
+  1. If an ED is defined for an asset in Asset Registry, return the ED as is.
+  
+  2. **Otherwise, pull a hard-coded ED from a `match` statement**.
+  
+  3. If no matches are found, return `Balance::MAX` which will prune unknown
+     assets.
+    
+It should be updated as follows:
+
+  1. If an ED is defined for an asset in Asset Registry, return the ED as is.
```

Comment:
In this case, it would be less straight forward to set ED to a desired value. However, I would still be achievable.
I don't have a strong opinion either way.

Reply:
Any place where we could cut down on runtime code/storage, we should. So let's get rid of it.

Reply:
I do not know what is better. Acala and Statemine do store ED. Acala does ED, ratio(in very complicated manner so) and DEX.

Reply:
ED protects from data (storage spam) and ratio from CPU spam.  Weights in substrate are moddelled by having sums of inpute data, storage calls and CPU.

Reply:
Statemine assets has callback which allows using ED as way to calculate fee but they do not force to do so (so if somebody has ratio - it can override)

Reply:
Also ED may be more or less static - to avoid accounts cleanup often. While ratios can vary to allow bias toward earning some currency (so you want more statble coint? make ratio small).

Reply:
Seems statemine is using a deterministic function to calculate ED https://github.com/paritytech/cumulus/blob/master/parachains/runtimes/assets/statemine/src/lib.rs#L217 
Better double check though

Reply:
> callback which allows using ED as way to calculate fee but they do not force to do so

Not sure how this is done, But I think I'm also ok with resolving this for now as we can change it anytime in the future without trouble.

Reply:
> Seems statemine is using a deterministic function to calculate ED https://github.com/paritytech/cumulus/blob/master/parachains/runtimes/assets/statemine/src/lib.rs#L217
@vimukthi-git can you update this link? I don't think it still points to where you were wanting it to

Reply:
It is here https://github.com/paritytech/cumulus/blob/3fb9c8a7beb8fb0fc31ea3335f8483d8086f60c8/parachains/runtimes/assets/statemine/src/constants.rs#L28 , seems someone just changed stuff around